src/appleseed/foundation/math/sampling/qmcsamplingcontext.h

```diff
@@ -381,6 +421,61 @@ inline Vector<T, N> QMCSamplingContext<RNG>::next2(Tag<Vector<T, N>>)
     return v;
 }
 
+template <typename RNG>
+template <typename T>
+inline Vector<T, 2> QMCSamplingContext<RNG>::next2(Tag<Vector<T, 2>>)
+{
+    Vector<T, 2> v;
+
+    assert(m_sample_count == 0 || m_instance < m_sample_count || m_cmj_params.m_enabled);
+    assert(m_dimension == 2);
+
+    if (m_mode == QMCMode)
+    {
+        if (m_cmj_params.m_enabled)
+        {
+            // Correlated Multi-Jittered Sampling
+            v = cmj_generate_sample<T>(
+                m_cmj_params.m_sequence++,
+                m_cmj_params.m_m,
+                m_cmj_params.m_n,
+                m_sample_count,
+                m_cmj_params.m_pattern);
+        }
+        else
+        {
+            if (m_instance < PrecomputedHaltonSequenceSize)
+            {
+                for (size_t i = 0; i < 2; ++i)
+                {
+                    v[i] = static_cast<T>(PrecomputedHaltonSequence[m_instance * 4 + i]);
+                    v[i] = rotate(v[i], static_cast<T>(m_offset[i]));
+                }
+            }
+            else
+            {
+                v[0] = radical_inverse_base2<T>(m_instance);
+                v[0] = rotate(v[0], static_cast<T>(m_offset[0]));
+
+                for (size_t i = 1; i < 2; ++i)
```

Comment:
That `for` loop is no longer necessary since the only possible value is `i` == 1.

Reply:
It was just to void copied lines.